test/functional/eval/environ_spec.lua

```diff
@@ -17,3 +22,63 @@ describe('environment variables', function()
     eq(0, exists('$DOES_NOT_EXIST'))
   end)
 end)
+
+describe('empty $HOME', function()
+  local original_home = ''
+
+  -- save $HOME before each test
+  before_each(function()
+    original_home = os.getenv('HOME')
+  end)
+
+  -- recover $HOME after each test
+  after_each(function()
+    if original_home ~= nil then
+      setenv('HOME', original_home)
+    end
+    os.remove('test_empty_home')
+  end)
+
+  local function tilde_in_pwd()
+    -- get files in pwd
+    command("let test_empty_home_pwd_files = split(globpath('.', '*'), '\n')")
+    -- get the index of the file named '~'
+    command('let test_empty_home_tilde_index = index(test_empty_home_pwd_files, "./~")')
+    return eval('test_empty_home_tilde_index') ~= -1
+  end
+
+  local function write_and_test_tilde()
+    system({nvim_prog, '-u', 'NONE', '-i', 'NONE', '--headless',
+                                          '-c', 'write test_empty_home', '+q'})
```

Comment:
could use `helpers.write_file`

Reply:
I think I am relying on nvim to create files in ~/.local/share/nvim/, with helpers.write_file, I might not be able to do that.